### PR TITLE
Get event

### PR DESCRIPTION
**Pull request by bmarty**
Try to get event faster after receiving a Push by retrieving the event directly. A background sync is run as usual.

Also give this event to the CallEventProcessor for faster incoming call treatment.

TODO:

- [x] Test all cases
- [x] Handle push rules
- [x] Ignore message edition
- [x] Do this also when app is in foreground? -> NO
- [x] Do not do this if not on WiFi network because of the network data overhead (the event will be downloaded 2 times)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/GetEventTask.kt [hunk 1]
```diff
@@ -16,28 +16,49 @@
...
             apiCall = roomAPI.getEvent(params.roomId, params.eventId)
         }
+
+        // Try to decrypt the Event
+        if (event.isEncrypted()) {
+            tryOrNull(message = "Unable to decrypt the event") {
+                eventDecryptor.decryptEvent(event, "")
+            }
+                    ?.let { result ->
+                        event.mxDecryptionResult = OlmDecryptionResult(
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/GetEventTask.kt:53]: Maybe we should do that directly from the eventDecryptor?

Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/GetEventTask.kt:53]: This is out of scope of this PR

**Review by ganfra — COMMENTED** [review]
I think we should offer the fastlane option to be used only on wifi in settings (enabled by default)
Otherwise LGTM, but didn't tested

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/events/DefaultEventService.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import timber.log.Timber
+import javax.inject.Inject
+
+internal class DefaultEventService @Inject constructor(
+        private val getEventTask: GetEventTask,
+        private val callEventProcessor: CallEventProcessor,
+        private val wifiDetector: WifiDetector
+) : EventService {
+
+    override suspend fun getEvent(roomId: String, eventId: String, onlyOnWifi: Boolean): Event? {
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/events/DefaultEventService.kt:33]: Also I'm not sure the onlyWifi should be exposed on the sdk...

Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/events/DefaultEventService.kt:33]: i wanted to give the ability to easily tell the SDK to save network data usage. But yes, I will move this check to the app level.

**Pull request merged**